### PR TITLE
Fix/vlad/affine to proj

### PR DESCRIPTION
**Pull request by vladfdp**
## Describe the changes

This PR fixes some of the affine -> projective functions by adding a condition
If the point in affine form is zero then return the projective zero

## Linked Issues

Resolves #

### CODE REVIEW

File: wrappers/golang/internal/generator/curves/templates/curve.go.tmpl [hunk 1]
```diff
@@ -43,9 +43,15 @@ func (p *{{.CurvePrefix}}Projective) FromAffine(a {{.CurvePrefix}}Affine) {{.Cur
 	z := {{.CurvePrefix}}BaseField{}
 	z.One()
 
+	if (a.X == z.Zero()) && (a.Y == z.Zero()) {
+		p.X = z.Zero()
+		p.Y = z.One()
+		p.Z = z.Zero()
```
Comment (jeremyfelder) [inline wrappers/golang/internal/generator/curves/templates/curve.go.tmpl:49]: Better to use the Zero function from the Projective point directly so we don't have multiple locations of its internal logic
```suggestion
		p.Zero()
```

File: wrappers/golang/internal/generator/curves/templates/curve.go.tmpl [hunk 2]
```diff
@@ -109,6 +115,14 @@ func (a *{{.CurvePrefix}}Affine) FromLimbs(x, y []uint32) {{.CurvePrefix}}Affine
 func (a {{.CurvePrefix}}Affine) ToProjective() {{.CurvePrefix}}Projective {
 	var z {{.CurvePrefix}}BaseField
 
+	if (a.X == z.Zero()) && (a.Y == z.Zero()) {
+		return {{.CurvePrefix}}Projective{
+			X: z.Zero(),
+			Y: z.One(),
+			Z: z.Zero(),
+		}
```
Comment (jeremyfelder) [inline wrappers/golang/internal/generator/curves/templates/curve.go.tmpl:123]: Same here as above
```suggestion
        var p {{.CurvePrefix}}Projective
		return p.Zero()
```

**Review by jeremyfelder — COMMENTED** [review]
Looks good for the fix.

I don't think it is too much work to extern the [from_affine](https://github.com/ingonyama-zk/icicle/blob/main/icicle/include/curves/projective.cuh#L30) function in cpp and use that directly in Rust and Go so we truly have a single implementation for this logic

**Review by LeonHibnik — APPROVED** [review]

**Comment by jeremyfelder** [comment]
>I don't think it is too much work to extern the [from_affine](https://github.com/ingonyama-zk/icicle/blob/main/icicle/include/curves/projective.cuh#L30) function in cpp and use that directly in Rust and Go so we truly have a single implementation for this logic

If we choose not to do this for the hotfix, then lets add a test case in both Rust and Go:
- Rust - https://github.com/ingonyama-zk/icicle/blob/main/wrappers/rust/icicle-core/src/tests.rs#L24
- Go
  - https://github.com/ingonyama-zk/icicle/blob/main/wrappers/golang/internal/generator/curves/templates/curve_test.go.tmpl#L38
  - https://github.com/ingonyama-zk/icicle/blob/main/wrappers/golang/internal/generator/curves/templates/curve_test.go.tmpl#L88

**Review by jeremyfelder — APPROVED** [review]

**Pull request merged**